src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -260,3 +265,116 @@ void QgsActiveLayerFeaturesLocatorFilter::triggerResult( const QgsLocatorResult
 
   QgisApp::instance()->mapCanvas()->zoomToFeatureIds( layer, QgsFeatureIds() << id );
 }
+
+//
+// QgsGotoLocatorFilter
+//
+
+QgsGotoLocatorFilter::QgsGotoLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{}
+
+void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  if ( feedback->isCanceled() )
+    return;
+
+  QgsLocatorResult result;
+  result.filter = this;
+  result.displayString = tr( "Go to %1" ).arg( string );
```

Comment:
I'm thinking - what about extracting the parsing code from triggerResult and also using it here, so that instead of "Go to http://www.openstreetmap.org/#map=6/46.423/4.746" you'd see "Go to 46.423° E, 4.746° N".

ALSO - if we yanked the parsing code out, we could do a check to see if the canvas CRS isn't 4326. And if not, then we:

1. Check if the coordinate is in the valid range for geographic coordinates
2. If so: emit two results, one for "Go to 11° E, 11° N" and one for "Go to 11, 11 (EPSG:3785)" (or whatever the canvas crs is)
3. If not: just emit one result "Go to 11, 11 (EPSG:3785)"

This would help confusion where users enter goto 55,55 expecting it to be in the project CRS, and instead get sent to a 4326 location.

(@NathanW2 when your stuff lands, we could refine this logic so we know if 11, 55 is valid for the current project CRS or if it MUST be 4326)